ta/pkcs11/src/processing_asymm.c

```diff
@@ -308,10 +333,75 @@ static enum pkcs11_rc load_tee_key(struct pkcs11_session *session,
 }
 
 static enum pkcs11_rc
-init_tee_operation(struct pkcs11_session *session __unused,
-		   struct pkcs11_attribute_head *proc_params __unused)
+init_tee_operation(struct pkcs11_session *session,
+		   struct pkcs11_attribute_head *proc_params,
+		   struct pkcs11_object *obj)
 {
-	return PKCS11_CKR_OK;
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+	struct active_processing *proc = session->processing;
+	struct rsa_pss_processing_ctx *rsa_pss_ctx = NULL;
+	size_t modulus_size = 0;
+	size_t hash_size = 0;
+	uint32_t k = 0;
+
+	switch (proc_params->id) {
+	case PKCS11_CKM_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA1_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA256_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA384_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA512_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA224_RSA_PKCS_PSS:
+		rc = pkcs2tee_proc_params_rsa_pss(proc, proc_params);
+		if (rc)
+			return rc;
+
+		rsa_pss_ctx = proc->extra_ctx;
+
+		switch (rsa_pss_ctx->hash_alg) {
+		case PKCS11_CKM_SHA_1:
+			hash_size = TEE_ALG_GET_DIGEST_SIZE(TEE_ALG_SHA1);
+			break;
+		case PKCS11_CKM_SHA224:
+			hash_size = TEE_ALG_GET_DIGEST_SIZE(TEE_ALG_SHA224);
+			break;
+		case PKCS11_CKM_SHA256:
+			hash_size = TEE_ALG_GET_DIGEST_SIZE(TEE_ALG_SHA256);
+			break;
+		case PKCS11_CKM_SHA384:
+			hash_size = TEE_ALG_GET_DIGEST_SIZE(TEE_ALG_SHA384);
+			break;
+		case PKCS11_CKM_SHA512:
+			hash_size = TEE_ALG_GET_DIGEST_SIZE(TEE_ALG_SHA512);
+			break;
+		default:
+			assert(0);
+			break;
+		}
+
+		modulus_size = get_object_key_bit_size(obj);
+
+		/**
+		 * The sLen field must be less than or equal to k*-2-hLen where
+		 * hLen is the length in bytes of the hash value. k* is the
+		 * length in bytes of the RSA modulus, except if the length in
+		 * bits of the RSA modulus is one more than a multiple of 8, in
+		 * which case k* is one less than the length in bytes of the
+		 * RSA modulus.
+		 */
+		if ((modulus_size % 8) == 0)
```

Comment:
`k = ROUNDUP(modulus_size, 8) / 8`

Reply:
OK

Reply:
Thanks for checking it out. Went for latter one as that follows the text more precisely.